nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -1803,6 +1833,206 @@ mod test {
         logctx.cleanup_successful();
     }
 
+    #[test]
+
+    fn test_disk_add_expunge_decommission() {
+        static TEST_NAME: &str = "planner_disk_add_expunge_decommission";
+        let logctx = test_setup_log(TEST_NAME);
+
+        // Create an example system with a single sled
+        let (example, blueprint1) =
+            ExampleSystemBuilder::new(&logctx.log, TEST_NAME).nsleds(1).build();
+        let mut collection = example.collection;
+        let input = example.input;
+
+        // The initial collection configuration has generation 1 for disks
+        // The initial blueprint configuration has generation 2 for disks
+        let (sled_id, disks_config) =
+            blueprint1.blueprint_disks.first_key_value().unwrap();
+        assert_eq!(disks_config.generation, Generation::from_u32(2));
+        assert_eq!(
+            collection
+                .sled_agents
+                .get(&sled_id)
+                .unwrap()
+                .omicron_physical_disks_generation,
+            Generation::new()
+        );
+
+        // All disks should have an `InService` disposition and `Active` state
+        for disk in &disks_config.disks {
+            assert_eq!(
+                disk.disposition,
+                BlueprintPhysicalDiskDisposition::InService
+            );
+            assert_eq!(disk.state, PhysicalDiskState::Active);
+        }
+
+        let mut builder = input.into_builder();
+
+        // Let's expunge a disk. It's disposition should change to `Expunged`
+        // but it's state should remain active.
```

Comment:
```suggestion
        // Let's expunge a disk. Its disposition should change to `Expunged`
        // but its state should remain active.
```

Reply:
Fixed in de4a0805309651eea73e6c8bab1e256afbaf992d